src/stax/ui_stax.c

```diff
@@ -0,0 +1,310 @@
+        (nbgl_obj_t *) passphrase_length_set_title(screenChildren[ICON_INDEX]);
+
+    // create nb words buttons
+    nbgl_objPoolGetArray(BUTTON, NB_BUTTONS, 0, (nbgl_obj_t **) &screenChildren[BUTTON_12_INDEX]);
+    passphrase_length_configure_buttons((nbgl_button_t **) &screenChildren[BUTTON_12_INDEX],
+                                        NB_BUTTONS,
+                                        (nbgl_touchCallback_t) &passphrase_length_callback);
+    ((nbgl_button_t *) screenChildren[BUTTON_12_INDEX])->text = passphraseLength[0];
+    ((nbgl_button_t *) screenChildren[BUTTON_18_INDEX])->text = passphraseLength[1];
+    ((nbgl_button_t *) screenChildren[BUTTON_24_INDEX])->text = passphraseLength[2];
+    ((nbgl_button_t *) screenChildren[BUTTON_24_INDEX])->borderColor = BLACK;
+    ((nbgl_button_t *) screenChildren[BUTTON_24_INDEX])->innerColor = BLACK;
+    ((nbgl_button_t *) screenChildren[BUTTON_24_INDEX])->foregroundColor = WHITE;
+
+    // create back button
+    screenChildren[BACK_BUTTON_INDEX] = (nbgl_obj_t *) passphrase_length_set_back_button(
+        (nbgl_touchCallback_t) &passphrase_length_callback);
+
+    nbgl_screenRedraw();
+}
+
+/*
+ * Word recover page
+ */
+static char textToEnter[MAX_WORD_LENGTH + 1] = {0};
+static int textIndex, suggestionIndex, keyboardIndex = 0;
+// the biggest word of BIP39 list is 8 char (9 with trailing '\0'), and
+// the max number of showed suggestions is NB_MAX_SUGGESTION_BUTTONS
+static char wordCandidates[(MAX_WORD_LENGTH + 1) * NB_MAX_SUGGESTION_BUTTONS] = {0};
+
+static void keyboard_dispatcher(const int token, uint8_t index __attribute__((unused))) {
+    if (token == BACK_BUTTON_TOKEN) {
+        nbgl_layoutRelease(layout);
+        if (remove_word_from_mnemonic()) {
+            display_keyboard_page();
+        } else {
+            passphrase_length_page();
+        }
+    } else if (token >= FIRST_SUGGESTION_TOKEN) {
+        nbgl_layoutRelease(layout);
+        PRINTF("Selected word is '%s' (size '%d')\n",
+               buttonTexts[token - FIRST_SUGGESTION_TOKEN],
+               strlen(buttonTexts[token - FIRST_SUGGESTION_TOKEN]));
+        add_word_in_mnemonic(buttonTexts[token - FIRST_SUGGESTION_TOKEN],
+                             strlen(buttonTexts[token - FIRST_SUGGESTION_TOKEN]));
+        if (is_mnemonic_complete()) {
+            display_result_page(check_mnemonic());
+        } else {
+            display_keyboard_page();
+        }
+    }
+}
+
+// function called when a key of keyboard is touched
+static void key_press_callback(const char touchedKey) {
+    size_t textLen = 0;
+    uint32_t mask = 0;
+    const size_t previousTextLen = strlen(textToEnter);
+    if (touchedKey == BACKSPACE_KEY) {
+        if (previousTextLen == 0) {
+            return;
+        }
+        textToEnter[previousTextLen - 1] = '\0';
+        textLen = previousTextLen - 1;
+    } else {
+        textToEnter[previousTextLen] = touchedKey;
+        textToEnter[previousTextLen + 1] = '\0';
```

Comment:
Yes, I always want the text to end with `\0` to be able to use `strlen` and such.
So `textToEnter` is defined as 
```
static char textToEnter[MAX_WORD_LENGTH + 1] = {0};
```
... so that there is always space for an extra `\0` character.